cpp/src/arrow/flight/flight_test.cc

```diff
@@ -207,6 +207,9 @@ ARROW_FLIGHT_TEST_ASYNC_CLIENT(GrpcAsyncClientTest);
 // Ad-hoc gRPC-specific tests
 
 TEST(TestFlight, ConnectUri) {
+#ifndef ARROW_ENABLE_THREADING
+  GTEST_SKIP() << "This test requires Boost.Process to run";
+#endif
```

Comment:
How about adding `arrow::testing::Process::IsAvailable()` static method or something and use it here?